federatedscope/core/auxiliaries/data_builder.py

```diff
@@ -118,6 +118,238 @@ def _generate_data(client_num=5,
+        data_list = []
+        for data_iter in dataset:
+            data, targets = [], []
+            if config.model.task == 'seq2seq':
+                for item in data_iter:
+                    data.append(
+                        vocab.get_vecs_by_tokens(tokenizer(item[1]),
+                                                 lower_case_backup=True))
+                    targets.append(
+                        vocab.get_vecs_by_tokens(tokenizer(item[0]),
+                                                 lower_case_backup=True))
+                targets = pad_sequence(targets).transpose(
+                    0, 1)[:, :raw_args['max_len'], :]
+            else:
+                for item in data_iter:
+                    data.append(
+                        vocab.get_vecs_by_tokens(tokenizer(item[1]),
+                                                 lower_case_backup=True))
+                    targets.append(item[0])
+                targets = label_to_index(targets)
+            data = pad_sequence(data).transpose(0,
+                                                1)[:, :raw_args['max_len'], :]
+            data_list.append([(x, y) for x, y in zip(data, targets)])
+
+        if len(data_list) == 3:
+            # Use raw splits
+            data_dict = {
+                'train': data_list[0],
+                'val': data_list[1],
+                'test': data_list[2]
+            }
+        elif len(data_list) == 2:
+            # Split train to (train, val)
+            data_dict = {
+                'train': data_list[0],
+                'val': None,
+                'test': data_list[1]
+            }
+            if splits:
+                train_size = int(splits[0] * len(data_dict['train']))
+                val_size = len(data_dict['train']) - train_size
+                lengths = [train_size, val_size]
+                data_dict['train'], data_dict[
+                    'val'] = torch.utils.data.dataset.random_split(
+                        data_dict['train'], lengths)
+        else:
+            # Use config.data.splits
+            data_dict = {}
+            train_size = int(splits[0] * len(data_list[0]))
+            val_size = int(splits[1] * len(data_list[0]))
+            test_size = len(data_list[0]) - train_size - val_size
+            lengths = [train_size, val_size, test_size]
+            data_dict['train'], data_dict['val'], data_dict[
+                'test'] = torch.utils.data.dataset.random_split(
+                    data_list[0], lengths)
+
+        return data_dict
+
+    def load_torchaudio_data(name, splits=None, config=None):
+        import torchaudio
+
+        dataset_func = getattr(import_module('torchaudio.datasets'), name)
+        raise NotImplementedError
+
+    def load_torch_geometric_data(name, splits=None, config=None):
+        import torch_geometric
+
+        dataset_func = getattr(import_module('torch_geometric.datasets'), name)
+        raise NotImplementedError
+
+    load_data = {
```

Comment:
It can be a global variable with the name `DATA_LOAD_FUNCS`, and be put in a more forward position.